util/oteltest/src/oteltest/__init__.py

```diff
@@ -0,0 +1,74 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import abc
+from typing import Mapping, Optional, Sequence
+
+from oteltest.telemetry import Telemetry
+
+
+class OtelTest(abc.ABC):
```

Comment:
I want to use oteltest to test a distro so I've created a base class that always install it so I don't have to repeat myself in each test case:

```
class MyBaseOtelTest(OtelTest):
    def requirements(self):
        return [f"-e {ROOT_DIR}"]
```

Unfortunately I cannot do without also defining all the other abstract methods. So what do you think on using a concrete class that raises NotImplementedError instead?

Reply:
Ended up with

```
class MyBaseOtelTest(OtelTest):
    def environment_variables(self):
        return {}

    def requirements(self):
        return [f"-e {ROOT_DIR}"]

    def wrapper_command(self):
        return ""

    def on_start(self):
        return None

    def on_stop(self, tel, stdout, stderr, returncode):
        raise NotImplementedError()
```

Reply:
Have you tried adding ABC as a parent of the intermediate abstract class? Like this:

```
class MyBaseOtelTest(OtelTest, ABC):
    def requirements(self):
        return [f"-e {ROOT_DIR}"]
```

Reply:
Yeah, that will probably work too